management/src/test/java/org/ehcache/management/cluster/ClusteringManagementServiceTest.java

```diff
@@ -241,7 +242,7 @@ public void test_stats_collection() throws Exception {
     assertThat(stats[0].getContext().get("cacheName"), equalTo("cache-1"));
     assertThat(stats[0].getStatistic(Counter.class, "GetCounter").getValue(), equalTo(4L));
     assertThat(stats[0].getContext().get("cacheName"), equalTo("cache-1"));
-    assertThat(stats[0].getStatistic(CounterHistory.class, "AllCacheGetCount").getValue().length, equalTo(1));
+    assertThat(stats[0].getStatistic(CounterHistory.class, "AllCacheGetCount").getValue().length, greaterThanOrEqualTo(1));
```

Comment:
I have actually seen this scenario happen with history samples.  You should make sure you grab the highest index from the array to ensure you get the most recent value.  You also need to wait for the value to initialized.  For instance, if the default value is 0, wait till its not 0 then grab it.  U need a check like belos (which is in a utility in my branch):

public static boolean isHistoryReady(AbstractStatisticHistory counterHistory, Long defaultValue) {
    if(counterHistory.getValue().length > 0) {
      int mostRecentIndex = counterHistory.getValue().length - 1;
      if((Long)counterHistory.getValue()[mostRecentIndex].getValue() > defaultValue) {
        return true;
      }
    }
    return false;
  }